client/app/vps/modal/backup-storage/edit-backup-storage.controller.js

```diff
@@ -27,14 +24,17 @@ class EditBackupStorageCtrl {
     }
 
     confirm () {
-        this.loader.save = true;
-        this.VpsService.putBackupStorageAccess(this.serviceName, this.row.ipBlock, this.model.ftp, this.model.nfs, this.model.cifs)
-            .then(() => this.CloudMessage.success(this.$translate.instant("vps_tab_backup_storage_set_success", {access: this.row.ipBlock})))
-            .catch(() => this.CloudMessage.error(this.$translate.instant("vps_tab_backup_storage_set_fail", {access: this.row.ipBlock})))
-            .finally(() => {
-                this.loader.save = false;
-                this.$uibModalInstance.close();
-            });
+        this.CloudMessage.flushChildMessage();
+        this.loader = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () => this.VpsService.putBackupStorageAccess(this.serviceName, this.row.ipBlock, this.model.ftp, this.model.nfs, this.model.cifs)
+                .then(() => this.CloudMessage.success(this.$translate.instant("vps_tab_backup_storage_set_success", { access: this.row.ipBlock })))
+                .catch(err => {
+                    this.CloudMessage.error(err.data.message);
```

Comment:
What if `err.data` is `undefined` ? And also is it necessary to have two messages for the same error ?